scripts/core/counterbasedevent.yml

```diff
@@ -0,0 +1,51 @@
+#
+# Copyright (C) 2019-2022 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+#
+# See YaML.md for syntax definition
+#
+--- #--------------------------------------------------------------------------
+type: header
+desc: "Intel $OneApi Level-Zero Extension APIs for Counter-based Events"
+version: "1.8"
+--- #--------------------------------------------------------------------------
+type: macro
+desc: "Counter-based Events Extension Name"
+version: "1.4"
+name: $X_EVENT_COUNTER_BASED_EXP_NAME
+value: '"$X_experimental_event_counter_based"'
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Counter-based Events Extension Version(s)"
+version: "1.8"
+name: $x_event_counter_based_exp_version_t
+etors:
+    - name: "1_0"
+      value: "$X_MAKE_VERSION( 1, 0 )"
+      desc: "version 1.0"
+--- #--------------------------------------------------------------------------
+type: enum
+desc: "Supported event flags for defining counter-based events. Must "
+class: $xEvent
+name: $x_event_counter_based_exp_flags_t
+etors:
+    - name: IMMEDIATE
+      desc: "Counter-based event is used for immediate command lists (default)"
+    - name: NON_REGULAR
+      desc: "Counter-based event is used for non-immediate command lists"
+--- #--------------------------------------------------------------------------
+type: struct
+desc: "Event descriptor for counter-based events. This structure may be passed to $xEventCreate as pNext member of $x_event_desc_t."
```

Comment:
do we want to consider adding flag for event pools for counter based events? i.e, would there be value in creating an event pool where events from that pool are only used as counting events?

Reply:
Yes we need flag for event memory pool as those events doesn't require pool memory to be allocated for them, the storage is inherited from command list.

Reply:
@MichalMrozek : sorry, could you clarify the comment about event pool? do we want the counter-based flags to be part of the zeEventPoolCreate or zeEventCreate, or both?

Reply:
Counter Based Events doesn't need memory backing if they are for immediate command lists, so this must be event pool flag as this is steered to control where allocation happens.
I do not see a need for event flag.
So only change in zeEventPoolCreate flags.

Reply:
seems to me that we are talking about "counter events" being associated at a command queue level. so would it be better to describe a different struct for this entirely so we can cleanly associate at command queue/command list level rather than shoe-horning it into existing event(pool) struct/desc?

we have fences described here as being associated at command queue level, so something similar for "counter events"?
https://spec.oneapi.io/level-zero/latest/core/PROG.html#fences

Reply:
We want to leverage events, otherwise all append API's would need to be rewritten.

Reply:
thanks @MichalMrozek .  I have moved the flags to the event pool logic. Please check.